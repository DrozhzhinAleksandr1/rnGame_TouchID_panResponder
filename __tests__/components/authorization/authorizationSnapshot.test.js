import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {AuthorizationScreen}  from '../../../app/components/authorization/authorizationscreen';
import stackNav from '../../../app/redux/reducers/stacknavreducer';
import { localization } from '../../../app/common/localization';

stackNav.setParams = () => {
    return "test";
}
mock = () =>{};

it('create snapshot Main showMenu={true}', () => {
    const renderer = new ShallowRenderer()
    let buttonRegistrationSnapshot = renderer.render(<AuthorizationScreen 
        CurrentLanguage={localization.ru} 
        navigation={stackNav} hideMenu={mock}
        loginError={false}
        passwordError={false}
        />);
    expect(buttonRegistrationSnapshot).toMatchSnapshot();
});