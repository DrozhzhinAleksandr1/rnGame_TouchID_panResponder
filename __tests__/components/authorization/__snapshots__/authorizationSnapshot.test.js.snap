// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot Main showMenu={true} 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#172533",
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <ScrollView
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "marginTop": "20%",
        }
      }
    >
      <Component
        style={
          Object {
            "alignItems": "center",
            "marginBottom": "10%",
          }
        }
      >
        <Input
          accessibilityLabel="LoginAuthorizationID"
          error={false}
          errorText="Ошибка ввода e-mail"
          filter={/\\^\\(\\(\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"\\]\\+\\(\\\\\\.\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"\\]\\+\\)\\*\\)\\|\\("\\.\\+"\\)\\)@\\(\\(\\\\\\[\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\]\\)\\|\\(\\(\\[a-zA-Z\\\\-0-9\\]\\+\\\\\\.\\)\\+\\[a-zA-Z\\]\\{2,\\}\\)\\)\\$/}
          placeholder="Email"
          testErrorID="LoginErrorAuthorizationID"
          testID="LoginAuthorizationID"
        />
        <Connect(InputPassWordWithSVG)
          placeholder="Пароль"
          warningStyle={
            Object {
              "borderColor": "#F86F6F",
            }
          }
        />
        <Component
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "width": 296,
            }
          }
        >
          <withNavigation(ButtonForgotPassword)
            text="Забыли пароль?"
          />
        </Component>
        <FormBtnSubmit
          accessibilityLabel="BtnLoginAuthorizationID"
          func={[Function]}
          isEnable={false}
          testID="BtnLoginAuthorizationID"
          text="Вход"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Roboto-Regular",
              "fontSize": 18,
              "marginTop": 30,
            }
          }
        >
          Ещё нет учётной записи?
        </Text>
        <withNavigation(ButtonRegistration)
          text="Зарегистрироваться"
        />
      </Component>
    </Component>
  </ScrollView>
  <Connect(ErrorValidation)
    testIDBtn="AuthorizationBtnId"
    testIDText="AuthorizationTextId"
  />
</Component>
`;
